chat-server/src/routes/conversations/createConversation.ts

```diff
@@ -20,51 +20,67 @@ export const CreateConversationRequest = SomeExpressRequest.extend({
   headers: z.object({
     "req-id": z.string(),
   }),
-  ip: z.string(),
-  origin: z.string(),
 });
 
 export interface CreateConversationRouteParams {
   conversations: ConversationsService;
+  createConversationCustomData?: AddCustomDataFunc;
 }
 
 export function makeCreateConversationRoute({
   conversations,
+  createConversationCustomData,
 }: CreateConversationRouteParams) {
   return async (
     req: ExpressRequest,
-    res: ExpressResponse<ApiConversation>,
+    res: ExpressResponse<ApiConversation, ConversationsRouterLocals>,
     next: NextFunction
   ) => {
     const reqId = getRequestId(req);
     try {
-      const { ip, origin: requestOrigin } = req;
-
-      if (!isValidIp(ip)) {
-        return sendErrorResponse({
+      logRequest({
+        reqId,
+        message: `Creating conversation`,
+      });
+      try {
```

Comment:
Can we un-nest this to use a single `try-catch`? This triple nested try block is smelly to me.

See how we do this in `addMessageToConversation`

Reply:
implemented. lmk what you think